docs/HelloWorldWalkthrough.md

```diff
@@ -0,0 +1,312 @@
+* The device must have a TPM that is enabled in the BIOS.
+* Ensure you are running on a device with Windows 11, version 10.0.26100.2454 or higher
+* Turn off [Secure Boot](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/disabling-secure-boot?view=windows-11)
+  * If Bitlocker is enabled, you will need access to your recovery keys
+* Enable [Test Signing](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/the-testsigning-boot-configuration-option). From an elevated command prompt, run the following and reboot: 
+  * ```BCDEdit /set TESTSIGNING ON```
+* Enable [Memory Integrity](https://learn.microsoft.com/en-us/windows/security/hardware-security/enable-virtualization-based-protection-of-code-integrity?tabs=security). In the Windows Security app, go to Device Security -&gt; Core Integrity -&gt; and toggle Memory Integrity ON, then Reboot.
+
+ 
+* Create a test certificate to use for signing the VBS Enclave. Replace "TheDefaultTestEnclaveCert" with the name you wish to create. eg:
+    ``` 
+    New-SelfSignedCertificate -CertStoreLocation Cert:\\CurrentUser\\My -DnsName "TheDefaultTestEnclaveCert" -KeyUsage DigitalSignature -KeySpec Signature -KeyLength 2048 -KeyAlgorithm RSA -HashAlgorithm SHA256 -TextExtension "2.5.29.37={text}1.3.6.1.5.5.7.3.3,1.3.6.1.4.1.311.76.57.1.15,1.3.6.1.4.1.311.97.814040577.346743379.4783502.105532346
+    ```
+
+## Solution structure
+This sample will create a Host app, a DLL project and include a solution level file as follows:
+- MyHostApp.sln
+  - MySecretEnclave.edl
+  - MyHostApp.vcxproj
+    - main.cpp
+  - MySecretEnclave.vcxproj
+    - MySecretEnclaveExports.cpp
+
+## Build the sample
+### Create the initial host app project and solution
+  1. Start VS, create new C++ Console App named **MyHostApp.vcxproj**
+### Add the Enclave Definition Language file to the solution
+  2. Add the Enclave Definition Language (EDL) file to the solution where you define your interface.
+     * Right click on solution, add new item, text file, named **MySecretEnclave.edl**
+     * Define a simple method for the enclave:
+        ```c
+        enclave
+        {
+            trusted
+            {
+                uint32_t DoSecretMath(
+                uint32_t val1,
+                uint32_t val2
+                );
+            };
+        };
+        ```
+### Add the Enclave DLL project to the solution
+Add a new DLL project and to the solution and configure it. This is the enclave dll.
+1. Right click add project, DLL Project, name it **MySecretVBSEnclave.vcxproj**
+2. Add references to the two nuget packages you created when you built the repro, located under \<repo root\>\\_build folder. Right-Click on the project, choose Manage Nuget packages – add a new package source for the _build folder where these were placed, and install the packages:
```

Comment:
>2. [](http://example.com/codeflow?start=0&length=2)

use 1. everywhere' for numbered lists

see: [https://github.com/microsoft/VbsEnclaveTooling/blob/main/README.md?plain=1](https://github.com/microsoft/VbsEnclaveTooling/blob/main/README.md?plain=1)

Reply:
that's weird, how do you reference/differentiate the steps?